lib/runner/job.ex

```diff
@@ -93,11 +86,13 @@ defmodule ElixirBench.Runner.Job do
     end
   end
 
-  defp get_benchmars_output_path(%Job{id: id}) do
+  @doc false
+  def get_benchmars_output_path(%Job{id: id}) do
     Confex.fetch_env!(:runner, :benchmars_output_path) <> "/" <> id
   end
 
-  defp get_compose_config(job) do
+  @doc false
+  def get_compose_config(job) do
     Antidote.encode!(%{version: "3", services: build_services(job)})
```

Comment:
so yeah see comments in tests but I generally like them better as private methods and then test it as part of where it is used (`start_job`) - making it public for testability should be the exception.

By making it public we couple our tests to their name, their parameter number and might also accidentally open them up for other people to use.

Keeping them private we only have one interface that is then usually more stable and can be abstracted better while we have the same confidence in it but more freedom to refactor its internals as long as we keep behaviour. I really love the chapter about this in Practical Object Oriented Design in Ruby by Sandi Metz

Reply:
See my comment above... I think we have an exception here, run_job logic is very complex.